# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [22]:
data['gender'].unique()
#учтем в нашем исследовании пол, для этого выведем уникальные значения из соответствующего столбца

array(['F', 'M', 'XNA'], dtype=object)

In [23]:
data['gender'].value_counts()#найдём количество тех, кто пол не указал или системную ошибку

F      14092
M       7238
XNA        1
Name: gender, dtype: int64

In [24]:
data = data[data['gender'] != 'XNA']#поскольку количество XNA (1) в таком массиве ни на что не повлияет, 
#то уберём эту строку

Исправил длинные строки здесь и далее внизу. Код работает, но читать разбитые строки, как в случае с формулой подсчета процента, общего для мужчин и женщин, не совсем удобно.

In [25]:
data.info()#проверим ещё раз на ошибки, количество данных соответствует "M" + "F"

<class 'pandas.core.frame.DataFrame'>
Int64Index: 21330 entries, 0 to 21524
Data columns (total 14 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   children               21330 non-null  int64  
 1   days_employed          21330 non-null  float64
 2   dob_years              21330 non-null  int64  
 3   education              21330 non-null  object 
 4   education_id           21330 non-null  int64  
 5   family_status          21330 non-null  object 
 6   family_status_id       21330 non-null  int64  
 7   gender                 21330 non-null  object 
 8   income_type            21330 non-null  object 
 9   debt                   21330 non-null  int64  
 10  total_income           21330 non-null  int64  
 11  purpose                21330 non-null  object 
 12  total_income_category  21330 non-null  object 
 13  purpose_category       21330 non-null  object 
dtypes: float64(1), int64(6), object(7)
memory usage: 2.4+ 

In [26]:
data_pivot_children = data.pivot_table(index='children', columns='gender', values='debt', aggfunc=['count', 'sum'])
display(data_pivot_children) # Сгруппируем в сводную таблицу очищенной data количество взятых кредитов ('count') 
#и просроченных ('sum') в зависимости от количества детей и пола 

count        sum     
gender       F     M    F    M
children                      
0         9520  4570  592  471
1         3085  1723  245  199
2         1256   796  134   60
3          196   134   17   10
4           28    13    1    3
5            7     2    0    0

При отображении с помощью функции display() ширина ячеек с названиями столбцов "поплыла". 
Еще проблема - появляется много свободного места в сводной таблице, и становится тяжело
соотнести данные с названиями столбцов
Я не до конца разобрался, как подстраивать ширину под содержимое и избавляться от пустых мест.

In [27]:
#созадим и выведем на экран долю не возвратных кредитов отдельно для женщин и мужчин и общую долю просрочек
#в зависимости от количества детей в семье
data_pivot_children['avg_debt_F'] = data_pivot_children['sum']['F'] / data_pivot_children['count']['F']
data_pivot_children['avg_debt_M'] = data_pivot_children['sum']['M'] / data_pivot_children['count']['M']
data_pivot_children['avg_debt_FM'] = (data_pivot_children['sum']['M'] + 
        data_pivot_children['sum']['F']) / (data_pivot_children['count']['M'] + 
            data_pivot_children['count']['F'])
display(data_pivot_children) 

count        sum      avg_debt_F avg_debt_M avg_debt_FM
gender       F     M    F    M                                  
children                                                        
0         9520  4570  592  471   0.062185   0.103063    0.075444
1         3085  1723  245  199   0.079417   0.115496    0.092346
2         1256   796  134   60   0.106688   0.075377    0.094542
3          196   134   17   10   0.086735   0.074627    0.081818
4           28    13    1    3   0.035714   0.230769    0.097561
5            7     2    0    0   0.000000   0.000000    0.000000

Вывод: С ростом числа детей количество просрочек ('debt') в целом падает, однако есть ряд нюансов.  
1. Выборка женщин с 4 и 5 детьми, а у мужчин ещё и с 3-мя - нерепрезентативны.
2. Cуммарно для женщин и мужчин (имеющих до 2х детей) доля невозрата немного возрастает с 7.5% до 9.5%
2. Женщины берут кредиты чаще в 1.5 - 2 раза.
3. Бездетных заёмщиков больше всего в выборке.
4. Бездетные женщины или женщины с одним ребенком более ответственны к погашению кредита чем мужчины тех же категорий
5. Отцы двух детей, наоборот, более ответственны, чем женщины.

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [28]:
data['family_status'].unique()
data['family_status'].value_counts()#найдем уникалные семейные статусы

женат / замужем          12261
гражданский брак          4133
Не женат / не замужем     2796
в разводе                 1189
вдовец / вдова             951
Name: family_status, dtype: int64

In [29]:
data_pivot_famstatus = data.pivot_table(index='family_status', columns='gender', values='debt', 
                                        aggfunc=['count', 'sum'])
display(data_pivot_famstatus) # Сгруппируем в сводную таблицу очищенной data количество взятых кредитов ('count') 
#и просроченных ('sum') в зависимости от количества детей и пола

count        sum     
gender                    F     M    F    M
family_status                              
Не женат / не замужем  1723  1073  118  155
в разводе               931   258   61   23
вдовец / вдова          896    55   52   11
гражданский брак       2832  1301  232  153
женат / замужем        7710  4551  526  401

In [30]:
#созадим и выведем на экран долю не возвратных кредитов отдельно для женщин и мужчин и общую долю для всех
#категорий семейного положения
data_pivot_famstatus['avg_debt_F'] = data_pivot_famstatus['sum']['F'] / data_pivot_famstatus['count']['F']
data_pivot_famstatus['avg_debt_M'] = data_pivot_famstatus['sum']['M'] / data_pivot_famstatus['count']['M']
data_pivot_famstatus['avg_debt_FM'] = (data_pivot_famstatus['sum']['M'] + 
        data_pivot_famstatus['sum']['F']) / (data_pivot_famstatus['count']['M'] + 
            data_pivot_famstatus['count']['F'])
display(data_pivot_famstatus) 

count        sum      avg_debt_F avg_debt_M avg_debt_FM
gender                    F     M    F    M                                  
family_status                                                                
Не женат / не замужем  1723  1073  118  155   0.068485   0.144455    0.097639
в разводе               931   258   61   23   0.065521   0.089147    0.070648
вдовец / вдова          896    55   52   11   0.058036   0.200000    0.066246
гражданский брак       2832  1301  232  153   0.081921   0.117602    0.093153
женат / замужем        7710  4551  526  401   0.068223   0.088113    0.075606

Вывод: Люди состоящие в браке более ответственные заёмщики, чем не состоящие в браке или не зарегистрировавшие брак. Однако и в этих данных есть особенности.  
1. Если учесть, что выборка вдовцов - нерепрезентативна, самыми безотстветственными заёмщиками следует признать неженатых мужчин.
2. Женщины в разводе и вдовы берут кредитов сильно больше, мужчин, и при этом являются одними из самых ответственных заемщиков судя по представленным данным. 
3. Как мужчины так женщины состоящие в неузаконенном (гражданском) браке - одни из наименее добросовестностных заёмщиков, исходя из полученных данных.
4. Регистрация брака практически не сказвается на добросоветность заемщиц, а у мужчин процент просрочки, наоборот, падает более чем в 1.5 раза.

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [31]:
data_pivot_incomecat = data.pivot_table(index='total_income_category', columns='gender', values='debt', 
                                        aggfunc=['count', 'sum'])
display(data_pivot_incomecat) # Сгруппируем в сводную таблицу очищенной data количество взятых кредитов ('count') 
#и просроченных ('sum') в зависимости от уровня дохода

count        sum     
gender                     F     M    F    M
total_income_category                       
A                         10    15    0    2
B                       2695  2318  162  192
C                      11063  4858  810  543
D                        308    41   15    6
E                         16     6    2    0

In [32]:
#созадим и выведем на экран долю не возвратных кредитов отдельно для женщин и мужчин и общую долю для всех
#категорий доходов
data_pivot_incomecat['avg_debt_F'] = data_pivot_incomecat['sum']['F'] / data_pivot_incomecat['count']['F']
data_pivot_incomecat['avg_debt_M'] = data_pivot_incomecat['sum']['M'] / data_pivot_incomecat['count']['M']
data_pivot_incomecat['avg_debt_FM'] = (data_pivot_incomecat['sum']['M'] + 
        data_pivot_incomecat['sum']['F']) / (data_pivot_incomecat['count']['M'] + 
                data_pivot_incomecat['count']['F'])
display(data_pivot_incomecat) 

count        sum      avg_debt_F avg_debt_M avg_debt_FM
gender                     F     M    F    M                                  
total_income_category                                                         
A                         10    15    0    2   0.000000   0.133333    0.080000
B                       2695  2318  162  192   0.060111   0.082830    0.070616
C                      11063  4858  810  543   0.073217   0.111774    0.084982
D                        308    41   15    6   0.048701   0.146341    0.060172
E                         16     6    2    0   0.125000   0.000000    0.090909

Вывод: С ростом уровнем дохода (те категории которые можно достоверно анализировать) количество просрочек ('debt') в целом меняется несущественно, хотя и 
1. В категории С можно заметить больше всего просрочек,
2. Судя по категориям В и С можно заметить, что мужчины - менее добросовестные заёмщики.  

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [33]:
data_pivot_purposecat = data.pivot_table(index='purpose_category', columns='gender', values='debt', 
                                         aggfunc=['count', 'sum'])
display(data_pivot_purposecat) # Сгруппируем в сводную таблицу очищенной data количество взятых кредитов ('count') 
#и просроченных ('sum') в зависимости от уровня дохода

count        sum     
gender                       F     M    F    M
purpose_category                              
операции с автомобилем    2843  1436  232  168
операции с недвижимостью  7046  3704  444  336
получение образования     2649  1339  206  163
проведение свадьбы        1554   759  107   76

In [34]:
#созадим и выведем на экран долю не возвратных кредитов отдельно для женщин и мужчин и общую долю для всех
#категорий по целям кредита
data_pivot_purposecat['avg_debt_F'] = data_pivot_purposecat['sum']['F'] / data_pivot_purposecat['count']['F']
data_pivot_purposecat['avg_debt_M'] = data_pivot_purposecat['sum']['M'] / data_pivot_purposecat['count']['M']
data_pivot_purposecat['avg_debt_FM'] = (data_pivot_purposecat['sum']['M'] + 
        data_pivot_purposecat['sum']['F']) / (data_pivot_purposecat['count']['M'] + 
            data_pivot_purposecat['count']['F'])
display(data_pivot_purposecat) 

count        sum      avg_debt_F avg_debt_M  \
gender                       F     M    F    M                         
purpose_category                                                       
операции с автомобилем    2843  1436  232  168   0.081604   0.116992   
операции с недвижимостью  7046  3704  444  336   0.063014   0.090713   
получение образования     2649  1339  206  163   0.077765   0.121733   
проведение свадьбы        1554   759  107   76   0.068855   0.100132   

                         avg_debt_FM  
gender                                
purpose_category                      
операции с автомобилем      0.093480  
операции с недвижимостью    0.072558  
получение образования       0.092528  
проведение свадьбы          0.079118

Вывод: Наименьший процент просрочек наблюдается в случае если заемщик берет кредит на операции с недвижимостью или на проведение свадьбы. Поскольку после анализа по полу все выборки остаются репрезентативные, то в анализе по эти параметрам можно заключить, что мужчины в среднем в 1,5 раза чаще имеют просрочки.

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

Ответ: Пропуски могут возникнуть из-за человеческого фактора: при внесении данных пользователь может не указать некоторые данные, особенно если поля не являются обязательными для заполнения. Пропуски также часто возникают из-за техничесих неисправностей, например при выгрузке/копировании данных может возникнуть сбой конечный массив будет неполным, с многочисленными 
пропусками. 

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

Ответ: поскольку медианное значение лучше характеризует выборку (её центр) чем среднее (особенно при наличии аномально больших/малых значений), поэтому логичнее заполнять пропуски значениями, которые не сместят центр выборки

### Шаг 4: общий вывод.

Напишите ваш общий вывод. Судя по имеющимся данным:
1. Женщины в 1,5 - 2 раза чаще берут кредиты
2. Наличие детей в целом плохо сказывается на доле погашения кредита без просрочек
3. Процент просрочек больше и у людей, не зарегистрировавших свой брак официально
4. Уровень дохода слабо влияет на % просрочек, и наиболее богатые и наиболее бедные категории в данных представлены нерепрезентативно.
5. Наименьший процент просрочек у клиентов, берущих деньги на операции с недвижимостью и на проведение свадеб.

До конца не понял что нужно сделать, но что-то похожее на красивый отчет оформить в самой последней ячейке этой тетради.

# Исследование надежности заёмщиков

В данной работе проводилось исследование надежности заёмщиков в зависимости от уровня доходов, целей займа
семейного положения и количества детей. Были получены результаты для женщин и мужчин, а также общие данные.

## По результатам исследования были выявлены следующие закономерности

#### Зависимость доли просроченных кредитов от уровня дохода:
    - Наиболее число просрочек наблюдается у заёмщиков с уровнем дохода 50000 - 200000
    - При этом мужчины менее добросоветные заёмщики
        
#### Зависимость доли просроченных кредитов от целей займа:
    - Наименьший процент просрочек наблюдается в случае если заемщик берет кредит на операции с недвижимостью 
    или на проведение свадьбы. 
    - Наибольший процент просрочек в категории "операции с авто".
    Разница просроченных займов достигает 2% в абсолютном выражении и до 20% относительно займов, связанных с
    недвижимостью
        
#### Зависимость доли просроченных кредитов от семейного положения заёмщика:
    - Как мужчины так женщины состоящие в неузаконенном (гражданском) браке - одни из наименее 
    добросовестностных заёмщиков.
    - Регистрация брака слабо сказвается на добросоветность заемщиц (% невозврата падает с 8.2 до 6.8), 
    а у официально женатых мужчин процент просрочки падает более чем в 1.5 раза по сравнению с официально 
    неженатыми мужчинами (с 12.5 до 8.8%). Отдельно выделяются холостяки, процент просрочек достигает 14.5%
    - Женщины в разводе и вдовы берут кредитов сильно больше, мужчин того же положения, но при этом являются
    одними из самых ответственных заемщиков.
        
#### Зависимость доли просроченных кредитов от семейного положения заёмщика:
    - В целом наибольшая доля просрочек наблюдается у заёмщиков с двумя детьми. 
    Доля невозрата у клиентов с 2 детьми заметно возрастает до 9.5%, против 7.5% у бездетных. 
    - Если же разделить заёмщиков по полу наблюдаются более интересные зависимости:
    
        - Бездетные женщины или женщины с одним ребенком более ответственны к погашению кредита, чем 
        мужчины тех же категорий
        - Отцы двух детей, наоборот, более ответственны, чем женщины.
